io-engine/src/lvm/property.rs

```diff
@@ -0,0 +1,190 @@
+//! Properties are attributes which are persisted in a given LVM resource and
+//! which can be used to identify or retrieve specific information from a
+//! resource, even across reboots.
+
+use crate::core::Protocol;
+use std::str::FromStr;
+
+#[macro_export]
+macro_rules! impl_properties {
```

Comment:
Probably, a crate already exists for this.
Serde would do, at least for some stuff.
If there's no crate suitable for the purpose, this code is better to be moved to a separate module.

Reply:
I'll add a ticket to research this